File: projects/finance/cartolas_test/parse.py
import pandas as pd
import sys

def analyze(file_path):
    print(f"--- Analyzing {file_path} ---")
    try:
        # Try to read as excel, if it fails maybe it's HTML
        df = pd.read_excel(file_path, header=None)
        
        # Find where the actual table starts (look for 'Monto', 'Cargo', etc.)
        for idx, row in df.iterrows():
            row_str = ' '.join([str(x).lower() for x in row.values])
            if 'monto' in row_str or 'cargo' in row_str or 'fecha' in row_str:
                print(f"Header found at row {idx}: {row.values}")
                print(df.iloc[idx:idx+15].to_string())
                break
        else:
            print("No clear header found. First 15 rows:")
            print(df.head(15).to_string())
    except Exception as e:
        print(f"Error reading with pandas read_excel: {e}")
        try:
            df = pd.read_html(file_path)[0]
            print("Successfully read as HTML. First 15 rows:")
            print(df.head(15).to_string())
        except Exception as e2:
            print(f"Error reading as HTML: {e2}")

analyze('Mov_Facturado.xls')
analyze('Saldo_y_Mov_No_Facturado.xls')

File: projects/finance/cartolas_test/parse.js
const XLSX = require('xlsx');

function parseXLS(file) {
    console.log(`\n--- parsing ${file} ---`);
    try {
        const workbook = XLSX.readFile(file);
        const sheet = workbook.Sheets[workbook.SheetNames[0]];
        const rows = XLSX.utils.sheet_to_json(sheet, { header: 1 });
        
        for (let i = 0; i < Math.min(rows.length, 30); i++) {
            console.log(`[${i}]: ${JSON.stringify(rows[i])}`);
        }
    } catch(e) {
        console.log("Error:", e);
    }
}

parseXLS('Mov_Facturado.xls');
parseXLS('Saldo_y_Mov_No_Facturado.xls');
